src/shipit_static_analysis/shipit_static_analysis/clang.py

```diff
@@ -43,6 +44,27 @@
     '--from-build', 'linux64-clang-tidy'
 ]
 
+__third_party = None
+
+
+def list_third_party(work_dir, path='tools/rewriting/ThirdPartyPaths.txt'):
+    '''
+    List third party directories using mozilla-central file
+    Uses a local cache to avoid reading every time the file
```

Comment:
What is the problem with reading the file every time? Current file systems (e.g. `ext4`) are already pretty good at caching files that are read often. Additionally, this local cache never seems to expire. Maybe it's better to just read the file every time?